### PR TITLE
Run functional tests every 2 hours and add another workflow that runs…

### PR DESCRIPTION
**Pull request by ytimocin**
… the tests on AKS instead of k3d

1. Run tests every 2 hours
2. Increase the VM sizes for the eshop-azure and eshop-aws tests
3. Add a new test workflow that runs every 2 hours and runs tests on AKS instead of k3d

Goal:
We will be collecting data for a week to see if the flakiness can be caused by k3d cluster.

### CODE REVIEW

File: .github/workflows/test-aks.yaml [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+          echo "Directories: $directories"
+          for index in "${!images[@]}"; do
+            image=${images[$index]}
+            directory=${directories[$index]}
+            echo "Building image $image from directory $directory"
+            docker build -t ghcr.io/radius-project/$image:${{ github.run_id }} $directory
+            docker tag ghcr.io/radius-project/$image:${{ github.run_id }} ghcr.io/radius-project/$image:${{ github.run_id }}
+            # We need to think about how we can delete these images at the end of the run
+            docker push ghcr.io/radius-project/$image:${{ github.run_id }}
+          done
```
Comment (ytimocin) [inline .github/workflows/test-aks.yaml]: This is where we build the necessary images and push them to GHCR. Would love to get ideas on this approach. I see that for some tests we are using the latest tag but for some we are building images. What would the best approach be here?

File: .github/workflows/test-aks.yaml [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+            sleep 60
+          done
+          aws eks update-kubeconfig --region ${{ env.AWS_REGION }} --name ${{ steps.gen-id.outputs.TEST_EKS_CLUSTER_NAME }}
+        timeout-minutes: 60
+        continue-on-error: false
+      - name: Install Dapr
+        if: steps.gen-id.outputs.RUN_TEST == 'true' && steps.gen-id.outputs.ENABLE_DAPR == 'true'
+        run: |
+          helm repo add dapr https://dapr.github.io/helm-charts/
+          helm upgrade --install dapr dapr/dapr --version=1.6 --namespace dapr-system --create-namespace --wait
```
Comment (ytimocin) [inline .github/workflows/test-aks.yaml]: Updated this so that if Dapr is installed there is no error.

File: .github/workflows/test-aks.yaml [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+            path: ./samples/eshop/eshop.bicep
+            uiTestFile: tests/eshop/eshop.app.spec.ts
+            credential: aws
+            enableDapr: false
+    env:
+      BRANCH: ${{ github.base_ref || github.ref_name }}
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      AZURE_LOCATION: westus3
+      AKS_RESOURCE_GROUP: samples-functional-tests
+      AKS_CLUSTER_NAME: samples-aks
```
Comment (youngbupark) [inline .github/workflows/test-aks.yaml]: @ytimocin If you use the existing aks cluster, workflow must clean up the existing resource first before starting the test. I recommend to add the code to delete namespace first before starting the test.

File: .github/workflows/test-aks.yaml [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+            --managed \
+            --region ${{ env.AWS_REGION }}
+          while [[ "$(eksctl get cluster ${{ steps.gen-id.outputs.TEST_EKS_CLUSTER_NAME }} --region ${{ env.AWS_REGION }} -o json | jq -r .[0].Status)" != "ACTIVE" ]]; do
+            echo "Waiting for EKS cluster to be created..."
+            sleep 60
+          done
+          aws eks update-kubeconfig --region ${{ env.AWS_REGION }} --name ${{ steps.gen-id.outputs.TEST_EKS_CLUSTER_NAME }}
+        timeout-minutes: 60
+        continue-on-error: false
+      - name: Install Dapr
```
Comment (youngbupark) [inline .github/workflows/test-aks.yaml]: If you use /test/azure/ bicep template, dapr is installed and maintained by AKS. So I would disable this one for AKS cluster.

File: 1 [hunk 5]
```diff
@@ -0,0 +1,17 @@
+Run functional tests every 2 hours and add another workflow that runs the tests on AKS instead of k3d
```
Comment (youngbupark) [inline 1]: You might check this in accidently.

**Review by youngbupark — APPROVED** [review]

**Pull request merged**